Parse/ParseNumber.java

```diff
@@ -25,4 +28,19 @@ public static void main(String[] args )
       }
       System.out.println(sb.toString());
    }
+   
+   static int ParseInt( char[] characters) throws NumberFormatException
+   {
+      int value = 0;
+      for( int i = 0; i < characters.length; i++ )
+      {
+         if( !Character.isDigit(characters[i]) )
+         {
+            throw new NumberFormatException();
+         }
+         value *=10;
+         value += characters[i] - '0';
```

Comment:
the digit characters have a numeric representation starting from 48 <- '0'; 49<-'1'; .. 57 <- '9'. Hence the expression characters[i] - '0' yield the current digit in the integer that is being constructed from its string representation!